spoolsense_installer/spoolman.py

```
@@ -3,51 +3,130 @@
-        ("filament", "dry_time_hours", "text", "Dry Time (hrs)"),
-    ]
-
-    for entity_type, key, field_type, display_name in fields:
+    if not _wait_for_spoolman(spoolman_url):
+        print(f"  {C.RED}✗{C.RESET} Spoolman not reachable at {spoolman_url} — could not create fields")
+        return [(entity_type, key) for entity_type, key, _, _ in EXTRA_FIELDS]
+
+    failed = []
+    for entity_type, key, field_type, display_name in EXTRA_FIELDS:
         try:
             req = urllib.request.Request(f"{spoolman_url}/api/v1/field/{entity_type}")
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)